packages/dataviews/src/dataforms-layouts/get-visible-fields.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Internal dependencies
+ */
+import { normalizeCombinedFields } from '../normalize-fields';
+import type {
+	Field,
+	CombinedFormField,
+	NormalizedCombinedFormField,
+} from '../types';
+
+export function getVisibleFields(
+	fields: Field< any >[],
+	formFields: string[] = [],
+	combinedFields?: CombinedFormField< any >[]
+): Field< any >[] {
+	const visibleFields: Array<
+		Field< any > | NormalizedCombinedFormField< any >
+	> = [ ...fields ];
+	if ( combinedFields ) {
+		visibleFields.push(
+			...normalizeCombinedFields( combinedFields, fields )
+		);
+	}
+	return formFields
+		.map( ( fieldId ) =>
+			visibleFields.find( ( { id } ) => id === fieldId )
+		)
+		.filter( ( field ): field is Field< any > => !! field );
+}
```

Comment:
I suggest maintaining consistency with the current typing by using `Item` instead of `any`.

```suggestion
export function getVisibleFields<Item>(
	fields: Field< Item >[],
	formFields: string[] = [],
	combinedFields?: CombinedFormField< Item >[]
): Field< Item >[] {
	const visibleFields: Array<
		Field< Item > | NormalizedCombinedFormField< Item >
	> = [ ...fields ];
	if ( combinedFields ) {
		visibleFields.push(
			...normalizeCombinedFields( combinedFields, fields )
		);
	}
	return formFields
		.map( ( fieldId ) =>
			visibleFields.find( ( { id } ) => id === fieldId )
		)
		.filter( ( field ): field is Field< Item > => !! field );
}
```

Reply:
Done as part of: https://github.com/WordPress/gutenberg/pull/65399/commits/9d44c7935a6b9c851292126a4d8280dabb412b8b